docs/sidecar/security.md

```diff
@@ -0,0 +1,628 @@
+
+```yaml
+# Example using External Secrets Operator
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: sidecar-secrets
+spec:
+  refreshInterval: 1h
+  secretStoreRef:
+    name: azure-keyvault
+    kind: SecretStore
+  target:
+    name: sidecar-secrets
+  data:
+  - secretKey: client-secret
+    remoteRef:
+      key: app-client-secret
+```
+
+## Signed HTTP Requests (SHR)
+
+SHR provides additional security by cryptographically binding tokens to HTTP requests.
+
+### When to Use SHR
+
+Use SHR for:
+- High-security scenarios requiring proof-of-possession
+- APIs that accept PoP tokens
+- Protection against token theft and replay attacks
+
+### Configuring SHR
+
+#### Generate Key Pair
+
+```bash
+# Generate RSA key pair
+openssl genrsa -out private.pem 2048
+openssl rsa -in private.pem -pubout -out public.pem
+
+# Base64 encode public key for configuration
+base64 -w 0 public.pem > public.pem.b64
+```
+
+#### Configure SHR in Kubernetes
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: shr-keys
+type: Opaque
+data:
+  public-key: <base64-encoded-public-key>
+  private-key: <base64-encoded-private-key>
+
+---
+containers:
+- name: sidecar
+  env:
+  - name: DownstreamApis__SecureApi__AcquireTokenOptions__PopPublicKey
+    valueFrom:
+      secretKeyRef:
+        name: shr-keys
+        key: public-key
+```
+
+#### Per-Request SHR
+
+```http
+GET /AuthorizationHeader/Graph?optionsOverride.AcquireTokenOptions.PopPublicKey=<base64-key> HTTP/1.1
+Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGc...
+```
+
+### SHR Best Practices
+
+1. **Protect Private Keys**: Store private keys securely and never expose them
+2. **Rotate Keys Regularly**: Implement key rotation policies
+3. **Validate PoP Tokens**: Ensure downstream APIs properly validate PoP tokens
+4. **Use Per-API Keys**: Different keys for different APIs improve isolation
+5. **Monitor Key Usage**: Audit and monitor SHR key usage
+
+## Token Security
+
+### Token Caching
+
+The sidecar caches tokens in memory by default:
+
+```yaml
+# In-memory cache (default)
+- name: TokenCache__EnableDistributedCache
+  value: "false"
+```
+
+For distributed scenarios with multiple replicas:
+
+```yaml
+# Distributed cache (Redis)
+- name: TokenCache__EnableDistributedCache
+  value: "true"
+- name: TokenCache__RedisConnection
+  value: "redis-host:6379,password=<redis-password>,ssl=true"
+```
+
+**Cache Security:**
+- Encrypt Redis connection (use SSL/TLS)
+- Use Redis authentication
+- Configure Redis network policies
+- Set appropriate token cache expiration
+- Monitor cache access patterns
+
+### Token Validation
+
+Configure strict token validation:
+
+```yaml
+- name: AzureAd__Audience
+  value: "api://<your-api-id>"  # Validate expected audience
+
+- name: AzureAd__Scopes
+  value: "access_as_user"  # Enforce required scopes
+```
+
+## Container Security
+
+### Run as Non-Root User
+
+Configure the sidecar to run as a non-root user:
```

Comment:
the container by default runs as non-root: https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web.Sidecar/Dockerfile.AzureLinux#L9

I don't think you need the runAsUser or runAsGroup below but pls verify

Reply:
@keegan-caruso. I'll merge the PR (as it looks pretty good)
I'll address more feedback (in particular if you disagree here)